src/NexusMods.App.UI/WorkspaceSystem/Workspace/WorkspaceViewModel.cs

```diff
@@ -352,8 +352,27 @@ private PageData GetDefaultPageData()
         return pageData;
     }
 
+    private bool TrySelectPage(PageData pageData)
+    {
+        var openTab = Panels
+            .Select(panel => (panel, tab: panel.Tabs.FirstOrDefault(tab => tab.Contents.PageData.Context.Equals(pageData.Context))))
+            .FirstOrOptional(tuple => tuple.tab is not null);
```

Comment:
In case of multiple versions of the same page being already open, this is picking the first one in what I would guess be panel layout order and then tab order.

If one panel has a selected tab matching the pageData while another panel has a hidden tab matching the page data, I would think the user would expect the already selected tab to become the target.

If there are multiple tabs in different panels, current panel selection should perhaps also be considered when deciding on the tab to select.

@captainsandypants Do you have rules in mind for which tab should be selected in case of multiple already existing tabs across different panels that already contain the desired Page?

Reply:
If you open a page with a standard left click, it should replace the content in the selected panel unless there is another panel open in the workspace with an active tab already displaying that page, in this case that panel should be selected. If there's a hidden tab with the same page as the one that was left clicked, it still follows the default behaviour, it doesn't need to know about hidden tabs. The left hand menu pages that are open in an active tab will be given the active state, if they are open in a hidden tab then they will have their default rest state i.e. nothing happens.

If you open a page with a right click and select open in new tab or panel, then it should always open in a new tab or panel, it does not need to know if that page is already open in a panel or any tab.

Reply:
Ah, I see. Yes, that would solve the hidden tab priority problem, thanks for the clarification!